{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pdfplumber\n",
    "pdf = pdfplumber.open(\"/home/hutianyi/PaperMill/论文工厂数据/论文工厂数据1/pdf/10.1002@jcb.26461.pdf\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "first_page = pdf.pages[0]\n",
    "tot_crop = first_page"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "second_page = pdf.pages[1]\n",
    "tot_crop = second_page"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "new_page = pdf.pages[3]\n",
    "tot_crop = new_page"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "找到正文对应的字号，只保留正文"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1273"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "pd.set_option ('display.max_rows', None)\n",
    "all_size = [round(x['size'], 2) for x in tot_crop.chars]\n",
    "all_font = [x['fontname'] for x in tot_crop.chars]\n",
    "all_text = [x['text'] for x in tot_crop.chars]\n",
    "df_size = pd.DataFrame({\n",
    "    'fontname': all_font,\n",
    "    'size': all_size,\n",
    "    'text': all_text\n",
    "    })\n",
    "''.join(all_text).index('Materials')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fontname</th>\n",
       "      <th>size</th>\n",
       "      <th>text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1273</th>\n",
       "      <td>TimesNewRomanPS-BoldMT</td>\n",
       "      <td>12.0</td>\n",
       "      <td>M</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1274</th>\n",
       "      <td>TimesNewRomanPS-BoldMT</td>\n",
       "      <td>12.0</td>\n",
       "      <td>a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1275</th>\n",
       "      <td>TimesNewRomanPS-BoldMT</td>\n",
       "      <td>12.0</td>\n",
       "      <td>t</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1276</th>\n",
       "      <td>TimesNewRomanPS-BoldMT</td>\n",
       "      <td>12.0</td>\n",
       "      <td>e</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1277</th>\n",
       "      <td>TimesNewRomanPS-BoldMT</td>\n",
       "      <td>12.0</td>\n",
       "      <td>r</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1278</th>\n",
       "      <td>TimesNewRomanPS-BoldMT</td>\n",
       "      <td>12.0</td>\n",
       "      <td>i</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1279</th>\n",
       "      <td>TimesNewRomanPS-BoldMT</td>\n",
       "      <td>12.0</td>\n",
       "      <td>a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1280</th>\n",
       "      <td>TimesNewRomanPS-BoldMT</td>\n",
       "      <td>12.0</td>\n",
       "      <td>l</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1281</th>\n",
       "      <td>TimesNewRomanPS-BoldMT</td>\n",
       "      <td>12.0</td>\n",
       "      <td>s</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1282</th>\n",
       "      <td>TimesNewRomanPS-BoldMT</td>\n",
       "      <td>12.0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1283</th>\n",
       "      <td>TimesNewRomanPS-BoldMT</td>\n",
       "      <td>12.0</td>\n",
       "      <td>a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1284</th>\n",
       "      <td>TimesNewRomanPS-BoldMT</td>\n",
       "      <td>12.0</td>\n",
       "      <td>n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1285</th>\n",
       "      <td>TimesNewRomanPS-BoldMT</td>\n",
       "      <td>12.0</td>\n",
       "      <td>d</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1286</th>\n",
       "      <td>TimesNewRomanPS-BoldMT</td>\n",
       "      <td>12.0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1287</th>\n",
       "      <td>TimesNewRomanPS-BoldMT</td>\n",
       "      <td>12.0</td>\n",
       "      <td>m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1288</th>\n",
       "      <td>TimesNewRomanPS-BoldMT</td>\n",
       "      <td>12.0</td>\n",
       "      <td>e</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1289</th>\n",
       "      <td>TimesNewRomanPS-BoldMT</td>\n",
       "      <td>12.0</td>\n",
       "      <td>t</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1290</th>\n",
       "      <td>TimesNewRomanPS-BoldMT</td>\n",
       "      <td>12.0</td>\n",
       "      <td>h</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1291</th>\n",
       "      <td>TimesNewRomanPS-BoldMT</td>\n",
       "      <td>12.0</td>\n",
       "      <td>o</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1292</th>\n",
       "      <td>TimesNewRomanPS-BoldMT</td>\n",
       "      <td>12.0</td>\n",
       "      <td>d</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1293</th>\n",
       "      <td>TimesNewRomanPS-BoldMT</td>\n",
       "      <td>12.0</td>\n",
       "      <td>s</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1294</th>\n",
       "      <td>TimesNewRomanPS-BoldMT</td>\n",
       "      <td>12.0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1295</th>\n",
       "      <td>TimesNewRomanPS-BoldItalicMT</td>\n",
       "      <td>12.0</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1296</th>\n",
       "      <td>TimesNewRomanPS-BoldItalicMT</td>\n",
       "      <td>12.0</td>\n",
       "      <td>l</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1297</th>\n",
       "      <td>TimesNewRomanPS-BoldItalicMT</td>\n",
       "      <td>12.0</td>\n",
       "      <td>i</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1298</th>\n",
       "      <td>TimesNewRomanPS-BoldItalicMT</td>\n",
       "      <td>12.0</td>\n",
       "      <td>n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1299</th>\n",
       "      <td>TimesNewRomanPS-BoldItalicMT</td>\n",
       "      <td>12.0</td>\n",
       "      <td>i</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1300</th>\n",
       "      <td>TimesNewRomanPS-BoldItalicMT</td>\n",
       "      <td>12.0</td>\n",
       "      <td>c</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1301</th>\n",
       "      <td>TimesNewRomanPS-BoldItalicMT</td>\n",
       "      <td>12.0</td>\n",
       "      <td>a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1302</th>\n",
       "      <td>TimesNewRomanPS-BoldItalicMT</td>\n",
       "      <td>12.0</td>\n",
       "      <td>l</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1303</th>\n",
       "      <td>TimesNewRomanPS-BoldItalicMT</td>\n",
       "      <td>12.0</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1304</th>\n",
       "      <td>TimesNewRomanPS-BoldItalicMT</td>\n",
       "      <td>12.0</td>\n",
       "      <td>s</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1305</th>\n",
       "      <td>TimesNewRomanPS-BoldItalicMT</td>\n",
       "      <td>12.0</td>\n",
       "      <td>a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1306</th>\n",
       "      <td>TimesNewRomanPS-BoldItalicMT</td>\n",
       "      <td>12.0</td>\n",
       "      <td>m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1307</th>\n",
       "      <td>TimesNewRomanPS-BoldItalicMT</td>\n",
       "      <td>12.0</td>\n",
       "      <td>p</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1308</th>\n",
       "      <td>TimesNewRomanPS-BoldItalicMT</td>\n",
       "      <td>12.0</td>\n",
       "      <td>l</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1309</th>\n",
       "      <td>TimesNewRomanPS-BoldItalicMT</td>\n",
       "      <td>12.0</td>\n",
       "      <td>e</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                          fontname  size text\n",
       "1273        TimesNewRomanPS-BoldMT  12.0    M\n",
       "1274        TimesNewRomanPS-BoldMT  12.0    a\n",
       "1275        TimesNewRomanPS-BoldMT  12.0    t\n",
       "1276        TimesNewRomanPS-BoldMT  12.0    e\n",
       "1277        TimesNewRomanPS-BoldMT  12.0    r\n",
       "1278        TimesNewRomanPS-BoldMT  12.0    i\n",
       "1279        TimesNewRomanPS-BoldMT  12.0    a\n",
       "1280        TimesNewRomanPS-BoldMT  12.0    l\n",
       "1281        TimesNewRomanPS-BoldMT  12.0    s\n",
       "1282        TimesNewRomanPS-BoldMT  12.0     \n",
       "1283        TimesNewRomanPS-BoldMT  12.0    a\n",
       "1284        TimesNewRomanPS-BoldMT  12.0    n\n",
       "1285        TimesNewRomanPS-BoldMT  12.0    d\n",
       "1286        TimesNewRomanPS-BoldMT  12.0     \n",
       "1287        TimesNewRomanPS-BoldMT  12.0    m\n",
       "1288        TimesNewRomanPS-BoldMT  12.0    e\n",
       "1289        TimesNewRomanPS-BoldMT  12.0    t\n",
       "1290        TimesNewRomanPS-BoldMT  12.0    h\n",
       "1291        TimesNewRomanPS-BoldMT  12.0    o\n",
       "1292        TimesNewRomanPS-BoldMT  12.0    d\n",
       "1293        TimesNewRomanPS-BoldMT  12.0    s\n",
       "1294        TimesNewRomanPS-BoldMT  12.0     \n",
       "1295  TimesNewRomanPS-BoldItalicMT  12.0    C\n",
       "1296  TimesNewRomanPS-BoldItalicMT  12.0    l\n",
       "1297  TimesNewRomanPS-BoldItalicMT  12.0    i\n",
       "1298  TimesNewRomanPS-BoldItalicMT  12.0    n\n",
       "1299  TimesNewRomanPS-BoldItalicMT  12.0    i\n",
       "1300  TimesNewRomanPS-BoldItalicMT  12.0    c\n",
       "1301  TimesNewRomanPS-BoldItalicMT  12.0    a\n",
       "1302  TimesNewRomanPS-BoldItalicMT  12.0    l\n",
       "1303  TimesNewRomanPS-BoldItalicMT  12.0     \n",
       "1304  TimesNewRomanPS-BoldItalicMT  12.0    s\n",
       "1305  TimesNewRomanPS-BoldItalicMT  12.0    a\n",
       "1306  TimesNewRomanPS-BoldItalicMT  12.0    m\n",
       "1307  TimesNewRomanPS-BoldItalicMT  12.0    p\n",
       "1308  TimesNewRomanPS-BoldItalicMT  12.0    l\n",
       "1309  TimesNewRomanPS-BoldItalicMT  12.0    e"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_size[1273:1310] "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(207.6661, 214.341042)"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tot_crop.chars[645]['x0'], tot_crop.chars[645]['x1']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(214.07603684, 217.86182484)"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tot_crop.chars[646]['x0'], tot_crop.chars[646]['x1']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(264.27558572, 267.09500152)"
      ]
     },
     "execution_count": 45,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tot_crop.chars[658]['x0'], tot_crop.chars[658]['x1']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(270.34181286, 273.48003186)"
      ]
     },
     "execution_count": 46,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tot_crop.chars[659]['x0'], tot_crop.chars[659]['x1']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "11.0    2107\n",
       "6.4        7\n",
       "Name: size, dtype: int64"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_size['size'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "SPOTBR+TimesNewRomanPSMT    1987\n",
       "SPOTBR+ErasDemiItalic        127\n",
       "Name: fontname, dtype: int64"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_size['fontname'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Cell Proliferation AssayTransfected cells for 48 h were seeded into the 96-well plate with 2000 cells per well. At 24 h, 48 h, 72 h and 96 h, cell counting kit-8 (CCK-8) re-agent (Dojindo Laboratories, Kumamoto, Japan) was supplied in each well. After 2 h of culture, absorbance of each well at 490 nm was recorded using a microplate reader.5-Ethynyl-2’-Deoxyuridine (EdU) Proliferation AssayTransfected cells for 24 h were labeled with 50 μM EdU (Beyotime, Shanghai, China) for 2 h, dyed with AdoLo and 4’,6-diamidino-2-phenylin-dole (DAPI) in dark for 30 min. EdU-positive rate was calculated as the ratio of EdU-positive cells (red) and DAPI-stained cells (blue) under a fluorescent microscope. Flow Cytometry Analysis of the Cell ApoptosisCells were washed with phosphate-buffered saline (PBS) twice, digested with ethylenedi-aminetetraacetic acid (EDTA)-free trypsin and resuspended at the density of 1×10/mL. Subse-quently, cells were incubated with 1× binding buf-fer and 1.25 μL of Annexin V-FITC (fluorescein isothiocyanate) for 15 min in dark. After centrif-ugation at 1000×g for 5 min, the precipitate was incubated with 10 μL of Propidium Iodide (PI). Apoptosis was determined within 1 hour by flow cytometry (Partec AG, Arlesheim, Switzerland).QRT-PCRWe extracted total RNA using TRIzol reagent (Invitrogen, Carlsbad, CA, USA), and reversely tran-scribed into the complementary deoxyribose nucleic acid (cDNA) using Primescript RT Reagent (TaKa-Ra, Otsu, Shiga, Japan). The qRT-PCR reaction was performed using SYBR Premix Ex TaqTM (TaKa-Ra, Otsu, Shiga, Japan) and StepOne Plus Real-time PCR System (Applied Biosystems, Foster City, CA, USA). Relative gene expression was calculated by 2 method. Primer sequences were listed as fol-lows: MiRNA-621: forward: 5’-TTAGCTCAGGAT-CATCATCATTTACATAGATAGGG-3’, reverse: 5’-AACACTCGAGTGAGAGAAGAGAGTG-CCTAGA-3’; U6: forward: 5’-CTCGCTTCGGCAG-CACA-3’, reverse: 5’-AACGCTTCACGAATTTG-CGT-3’; SIX4: forward: 5’-TCTCGGGGTGATC-GACAAGAA-3’, reverse: 5’-CCCTTTGTTCAT-TCGTTCCTGG-3’; β-actin: forward: 5›-CCTGG-CACCCAGCACAAT-3›, reverse: 5›-GCTGATCCA-CATCTGCTGGAA-3›.'"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "''.join(df_size[df_size['size'] == 11]['text'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "''"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "''.join(df_size[df_size['fontname'] == 'OFALOU+ErasDemi']['text'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "FONT_SIZE = {\n",
    "            'chapter' : [10, 12],\n",
    "            'section' : [],\n",
    "            'maintext' : [10, 10],\n",
    "}\n",
    "FONT_NAME = {\n",
    "    'chapter' : ['MKJLCA', 'MKJLCA'],\n",
    "    'section' : [],\n",
    "    'maintext' : ['MKJKPP', 'MKJLAA'],\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "class Section:\n",
    "    def __init__(self, name):\n",
    "        self.name = name\n",
    "        self.text = ''\n",
    "    def add_text(self, text):\n",
    "        self.text += text\n",
    "    def print_text(self):\n",
    "        print('###', self.name)\n",
    "        print(self.text)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "class Chapter:\n",
    "    def __init__(self, name):\n",
    "        self.name = name\n",
    "        self.sections = []\n",
    "    def get_cursec(self):\n",
    "        if len(self.sections) == 0:\n",
    "            self.sections.append(Section('Head Section'))\n",
    "        return self.sections[-1]\n",
    "    def add_sec(self, name):\n",
    "        self.sections.append(Section(name))\n",
    "    def print_sections(self):\n",
    "        sections = self.sections\n",
    "        print('##', self.name)\n",
    "        # print('   All Sections', [x.name for x in sections])\n",
    "        for index, section in enumerate(sections):\n",
    "            section.print_text()\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "from math import fabs\n",
    "class Article:\n",
    "    def __init__(self, name):\n",
    "        self.name = name\n",
    "        self.chapters = []  \n",
    "    def equal(self, x, y):\n",
    "        EPS = 0.01\n",
    "        if fabs(x -y) < EPS:\n",
    "            return True\n",
    "        else:\n",
    "            return False\n",
    "    def read_text(self, chars, index, fontname, fontsize):\n",
    "        text = ''\n",
    "        while index < len(chars):\n",
    "            if chars[index]['fontname'] == fontname and self.equal(chars[index]['size'], fontsize):\n",
    "                text += chars[index]['text']\n",
    "            else:\n",
    "                break\n",
    "            index += 1\n",
    "        return index, text\n",
    "    def judge(self, char):\n",
    "        FONT_SIZE ={\n",
    "            'chapter' : [12, 11.4],\n",
    "            'section' : [11, 9.4],\n",
    "            'maintext' : [11, 11, 9.4],\n",
    "        }\n",
    "        FONT_NAME ={\n",
    "            'chapter' : ['OFALOU+ErasDemi', 'OFALOU+ErasUltra'],\n",
    "            'section' : ['OFALOU+ErasDemiItalic', 'OFALOU+Frutiger-Black'],\n",
    "            'maintext' : ['OFALOU+TimesNewRomanPSMT', 'OFALOU+TimesNewRomanPS-ItalicMT', 'OFALOU+Helvetica-Bold'],\n",
    "        }\n",
    "        fontname = char['fontname']\n",
    "        fontsize = char['size']\n",
    "        for type_name in ['chapter', 'section', 'maintext']:\n",
    "            namelist = FONT_NAME[type_name]\n",
    "            sizelist = FONT_SIZE[type_name]\n",
    "            for i in range(len(namelist)):\n",
    "                if fontname == namelist[i] and equal(fontsize, sizelist[i]):\n",
    "                    return type_name\n",
    "        return 'none'\n",
    "    def parse(self, pdf_crop):\n",
    "        index = 0\n",
    "        while index < len(pdf_crop.chars):\n",
    "            char = pdf_crop.chars[index]\n",
    "            type_name = self.judge(char)\n",
    "            if type_name == 'chapter':\n",
    "                index, text = self.read_text(pdf_crop.chars, index, char['fontname'] ,char['size'])\n",
    "                self.add_chap(text)\n",
    "            elif type_name == 'section':\n",
    "                index, text = self.read_text(pdf_crop.chars, index, char['fontname'] ,char['size'])\n",
    "                self.add_sec(text)\n",
    "            elif type_name ==  'maintext':\n",
    "                index, text = self.read_text(pdf_crop.chars, index, char['fontname'] ,char['size'])\n",
    "                self.add_text(text)\n",
    "            else:\n",
    "                index += 1\n",
    "    def get_curchap(self):\n",
    "        if len(self.chapters) == 0:\n",
    "            self.chapters.append(Chapter('Head Chapter'))\n",
    "        return self.chapters[-1]\n",
    "    def add_chap(self, name):\n",
    "        self.chapters.append(Chapter(name))\n",
    "    def add_sec(self, text):\n",
    "        chap = self.get_curchap()\n",
    "        sec = chap.add_sec(text)\n",
    "    def add_text(self, text):\n",
    "        chap = self.get_curchap()\n",
    "        sec = chap.get_cursec()\n",
    "        sec.add_text(text)\n",
    "    def print_chapters(self):\n",
    "        chapters = self.chapters\n",
    "        print('#', self.name)\n",
    "        # print('# All Chapters', [x.name for x in chapters])\n",
    "        for index, chapter in enumerate(chapters):\n",
    "            chapter.print_sections()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [],
   "source": [
    "def read_text(chars, index, fontname, fontsize):\n",
    "    text = ''\n",
    "    while index < len(chars):\n",
    "        if chars[index]['fontname'] == fontname and equal(chars[index]['size'], fontsize):\n",
    "            text += chars[index]['text']\n",
    "        else:\n",
    "            break\n",
    "        index += 1\n",
    "    return index, text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "from math import fabs\n",
    "def equal(x, y):\n",
    "    EPS = 0.11\n",
    "    if fabs(x -y) < EPS:\n",
    "        return True\n",
    "    else:\n",
    "        return False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [],
   "source": [
    "article = Article('Mitochondrial protein UCP1 mediates liver injury induced by LPS through EKR signaling pathway')\n",
    "index = 0\n",
    "while index < len(tot_crop.chars):\n",
    "    char = tot_crop.chars[index]\n",
    "    if char['fontname'] == FONT_NAME['chapter'] and equal(char['size'], FONT_SIZE['chapter']):\n",
    "        index, text = read_text(tot_crop.chars, index, char['fontname'] ,char['size'])\n",
    "        article.add_chap(text)\n",
    "    elif char['fontname'] == FONT_NAME['section'] and equal(char['size'], FONT_SIZE['section']):\n",
    "        index, text = read_text(tot_crop.chars, index, char['fontname'] ,char['size'])\n",
    "        article.add_sec(text)\n",
    "    elif char['fontname'] == FONT_NAME['maintext'] and equal(char['size'], FONT_SIZE['maintext']):\n",
    "        index, text = read_text(tot_crop.chars, index, char['fontname'] ,char['size'])\n",
    "        article.add_text(text)\n",
    "    else:\n",
    "        index += 1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[]"
      ]
     },
     "execution_count": 36,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "[x.name for x in article.chapters]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "article = Article('test')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Experimental Model',\n",
       " 'Liver Tissue Preparation',\n",
       " 'Caspase-3 Activity Detection',\n",
       " 'ATP Production']"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "[x.name for x in chap2.sections]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'in mitochondrial function, which may be closely related to liver injury. In this study, we intend to explore the potential relationship between UCP1 le-vels in liver cells and liver injury induced by LPS.As there is still lack of evidence about the relation-ship between UCP1 and LPS induced liver injury, we selected patients with liver injury from ICU and tested UCP1 level in the liver based on the data con-cerning mouse model of liver injury, in order to eva-luate the value of UCP1 as molecular biomarker in liver injury induced by LPS and provide theoretical basis for the early diagnosis and prognosis of liver injury.'"
      ]
     },
     "execution_count": 73,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sec1.text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Experimental Model',\n",
       " 'Liver Tissue Preparation',\n",
       " 'Caspase-3 Activity Detection',\n",
       " 'ATP Production']"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chap2 = article.chapters[1]\n",
    "sections = chap2.sections\n",
    "[x.name for x in sections]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Mouse liver injury model was established using LPS according to previous report. Female C57BL/6 mouse at 6-8 weeks old was intraperi-toneal injected by LPS at 25 μg/kg. The mouse injected by normal saline was treated as control. The experiment was approved by the Ethics Com-mittee in our hospital conforming to animal wel-fare. The liver tissue collected from the sepsis patients received surgery was used as experimen-tal group, while the liver tissue obtained from li-ver operation due to traffic accident was treated as control. The experiment was approved by the Ethics Committee in Qingdao Municipal Hospi-tal. All the subjects had signed informed consent.'"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sections[0].text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'The liver tissue was extracted from the mouse or operation. The tissue was lysed and washed by normal saline and acetone for four times. Then, the homogenate was centrifuged at 150 g and sto-red at -20°C.'"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sections[1].text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'ERK signaling activity was detected by kinase activity kit. The caspase-3 activity was tested by microplate reader. A total of 2 × 10 liver cel-ls were added with lysate to release the protein. Then, the mixture was incubated with caspase-3 chromophoric substance at room temperature avoid of light for 20 min. At last, cells were tested on microplate reader at 492 nm.'"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sections[2].text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Mitochondrial ATP production in liver tissue was detected by flow cytometry. The liver tissue was added with lysis to prepare the suspension. '"
      ]
     },
     "execution_count": 69,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sections[3].text"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "article.parse(tot_crop)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# test\n"
     ]
    }
   ],
   "source": [
    "article.print_chapters()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "article1.parse(right_crop)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Article Name Mitochondrial protein UCP1 mediates liver injury induced by LPS through EKR signaling pathway\n",
      "# All Chapters ['Head Chapter', 'Materials and Methods', 'Results']\n",
      "1## Chapter Name Head Chapter\n",
      "   All Sections ['Head Section']\n",
      "1### Section Name Head Section\n",
      "    in mitochondrial function, which may be closely related to liver injury. In this study, we intend to explore the potential relationship between UCP1 le-vels in liver cells and liver injury induced by LPS.As there is still lack of evidence about the relation-ship between UCP1 and LPS induced liver injury, we selected patients with liver injury from ICU and tested UCP1 level in the liver based on the data con-cerning mouse model of liver injury, in order to eva-luate the value of UCP1 as molecular biomarker in liver injury induced by LPS and provide theoretical basis for the early diagnosis and prognosis of liver injury.\n",
      "2## Chapter Name Materials and Methods\n",
      "   All Sections ['Experimental Model', 'Liver Tissue Preparation', 'Caspase-3 Activity Detection', 'ATP Production', 'Western Blot', 'Real-time PCR', 'U0126 Pretreatment', 'Statistical Analysis']\n",
      "1### Section Name Experimental Model\n",
      "    Mouse liver injury model was established using LPS according to previous report. Female C57BL/6 mouse at 6-8 weeks old was intraperi-toneal injected by LPS at 25 μg/kg. The mouse injected by normal saline was treated as control. The experiment was approved by the Ethics Com-mittee in our hospital conforming to animal wel-fare. The liver tissue collected from the sepsis patients received surgery was used as experimen-tal group, while the liver tissue obtained from li-ver operation due to traffic accident was treated as control. The experiment was approved by the Ethics Committee in Qingdao Municipal Hospi-tal. All the subjects had signed informed consent.\n",
      "2### Section Name Liver Tissue Preparation\n",
      "    The liver tissue was extracted from the mouse or operation. The tissue was lysed and washed by normal saline and acetone for four times. Then, the homogenate was centrifuged at 150 g and sto-red at -20°C.\n",
      "3### Section Name Caspase-3 Activity Detection\n",
      "    ERK signaling activity was detected by kinase activity kit. The caspase-3 activity was tested by microplate reader. A total of 2 × 10 liver cel-ls were added with lysate to release the protein. Then, the mixture was incubated with caspase-3 chromophoric substance at room temperature avoid of light for 20 min. At last, cells were tested on microplate reader at 492 nm.\n",
      "4### Section Name ATP Production\n",
      "    Mitochondrial ATP production in liver tissue was detected by flow cytometry. The liver tissue was added with lysis to prepare the suspension. Next, the mixture was added with detection flu-id and tested on microplate reader at 560 nm to calculate the ATP concentration according to the standard curve.\n",
      "5### Section Name Western Blot\n",
      "    The protein was extracted from the liver tissue and quantified. Next, the protein was separated by SDS-PAGE and analyzed by ImageJ software to obtain the gray value.\n",
      "6### Section Name Real-time PCR\n",
      "    Total RNA was extracted from the liver tissue using Trizol (Invitrogen, Carlsbad, CA, USA) and reverse transcribed for Real-time PCR. To-tal RNA was reverse transcribed to cDNA using polyA at 42°C for 60 min. The PCR reaction system contained 1 μl cDNA, 2 μl primers, 4 μl dNTP, 2 μl loading buffer, and 1 μl Taq enzyme. The PCR reaction was composed of 94°C for 5 min, followed by 30 cycles of 94°C for 30 s and 55°C for 30 s. The primer sequences were as follows.UCP1, 5’-TTATTAGAGGGTGGGGTGGATTGT-3’ and 5’- CCACCTAAATCAACCTCCAACCA-3’; actin, 5’-TTATTAGAGGGTGGGGCGGATCGC-3’ and 5’- CCACCTAAATCGACCTCCGACCG-3’.\n",
      "7### Section Name U0126 Pretreatment\n",
      "    ERK signaling pathway inhibitor U0126 was used to treat mouse through intraperitoneal injection at 10 μg/kg. Next, the mouse was intra-peritoneal injected by LPS at 25 μg/kg. The mouse injected by normal saline was treated as control.\n",
      "8### Section Name Statistical Analysis\n",
      "    All data analyses were performed on SPSS 11.0 (SPSS Inc., Chicago, IL, USA). The measurement data were depicted as mean ± standard deviation. p<0.05 was considered as statistical significance.\n",
      "3## Chapter Name Results\n",
      "   All Sections ['UCP1 Expression in Liver Cells from Mouse Liver Injury Model']\n",
      "1### Section Name UCP1 Expression in Liver Cells from Mouse Liver Injury Model\n",
      "    As shown in Figure 1A, qRT-PCR results demonstrated that UCP1 mRNA significant-ly increased in liver cells from liver injury model compared with that in normal control (p=0.0017). As shown in Figure 1B and C, We-stern blot results revealed that the level of UCP1 protein obviously elevated in liver cells from li-ver injury model compared with that in normal control (p=0.0027).\n"
     ]
    }
   ],
   "source": [
    "article1.print_chapters()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CP1 mediates \n",
      "LPS through \n",
      "G4, G.-J. SHI5\n",
      "icipal Hospital, Qingdao, Shandong, China\n",
      "filiated Hospital of Shihezi University Medical \n",
      " Xinjiang, China\n",
      "g People’s Hospital, Qingdao, Shandong, China\n",
      "ospital, Qingdao, Shandong, China\n",
      "unicipal Hospital, Qingdao, Shandong, China\n",
      "D\n",
      "Introduction\n",
      "E\n",
      "LPS induced liver injury, which belongs to a kind \n",
      "of liver inflammatory response syndrome. It is cha-\n",
      "T\n",
      "racterized by high incidence, high case fatality rate, \n",
      "high cost, and increasing incidence1,2. Liver in\n"
     ]
    }
   ],
   "source": [
    "x_edge = 0.07\n",
    "y_edge = 0.1\n",
    "half = 0.5\n",
    "height = first_page.height\n",
    "width = first_page.width\n",
    "left_crop = first_page.within_bbox((width * x_edge, height * y_edge, width * half, height * (1 - y_edge)))\n",
    "right_crop = first_page.within_bbox((width * half, height * y_edge, width * (1 - x_edge), height * (1 - y_edge)))\n",
    "\n",
    "print(right_crop.extract_text()[:500])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>fontname</th>\n",
       "      <th>size</th>\n",
       "      <th>text</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>OFALOU+ErasUltra</td>\n",
       "      <td>11.4</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>OFALOU+ErasUltra</td>\n",
       "      <td>11.4</td>\n",
       "      <td>b</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>OFALOU+ErasUltra</td>\n",
       "      <td>11.4</td>\n",
       "      <td>s</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>OFALOU+ErasUltra</td>\n",
       "      <td>11.4</td>\n",
       "      <td>t</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>OFALOU+ErasUltra</td>\n",
       "      <td>11.4</td>\n",
       "      <td>r</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>OFALOU+ErasUltra</td>\n",
       "      <td>11.4</td>\n",
       "      <td>a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>OFALOU+ErasUltra</td>\n",
       "      <td>11.4</td>\n",
       "      <td>c</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>OFALOU+ErasUltra</td>\n",
       "      <td>11.4</td>\n",
       "      <td>t</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>OFALOU+ErasUltra</td>\n",
       "      <td>11.4</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>OFALOU+Frutiger-Black</td>\n",
       "      <td>9.4</td>\n",
       "      <td>–</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>OFALOU+Frutiger-Black</td>\n",
       "      <td>9.4</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>OFALOU+Frutiger-Black</td>\n",
       "      <td>9.4</td>\n",
       "      <td>O</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>OFALOU+Frutiger-Black</td>\n",
       "      <td>9.4</td>\n",
       "      <td>B</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>OFALOU+Frutiger-Black</td>\n",
       "      <td>9.4</td>\n",
       "      <td>J</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>OFALOU+Frutiger-Black</td>\n",
       "      <td>9.4</td>\n",
       "      <td>E</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>OFALOU+Frutiger-Black</td>\n",
       "      <td>9.4</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>OFALOU+Frutiger-Black</td>\n",
       "      <td>9.4</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>OFALOU+Frutiger-Black</td>\n",
       "      <td>9.4</td>\n",
       "      <td>I</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>OFALOU+Frutiger-Black</td>\n",
       "      <td>9.4</td>\n",
       "      <td>V</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>OFALOU+Frutiger-Black</td>\n",
       "      <td>9.4</td>\n",
       "      <td>E</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>OFALOU+Frutiger-Black</td>\n",
       "      <td>9.4</td>\n",
       "      <td>:</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>M</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>i</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>t</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>o</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>c</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>h</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>o</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>d</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>r</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>i</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>r</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>e</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>b</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>u</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>d</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>t</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>i</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>l</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>i</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>v</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>e</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>r</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>.</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>58</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>59</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>h</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>61</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>e</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>62</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>63</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>r</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>o</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>65</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>l</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>66</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>e</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>67</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>s</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>o</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>f</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>71</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>72</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>m</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>73</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>i</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>t</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>o</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>76</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>c</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>77</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>h</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>o</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>79</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>80</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>d</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>81</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>r</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>i</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>83</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>a</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>84</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>l</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>p</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>87</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>r</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>88</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>o</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>89</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>t</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>e</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>91</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>i</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>92</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>93</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>94</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>i</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>95</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>n</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>97</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>l</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>98</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>i</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>99</th>\n",
       "      <td>OFALOU+Helvetica-Bold</td>\n",
       "      <td>9.4</td>\n",
       "      <td>v</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                 fontname  size text\n",
       "0        OFALOU+ErasUltra  11.4    A\n",
       "1        OFALOU+ErasUltra  11.4    b\n",
       "2        OFALOU+ErasUltra  11.4    s\n",
       "3        OFALOU+ErasUltra  11.4    t\n",
       "4        OFALOU+ErasUltra  11.4    r\n",
       "5        OFALOU+ErasUltra  11.4    a\n",
       "6        OFALOU+ErasUltra  11.4    c\n",
       "7        OFALOU+ErasUltra  11.4    t\n",
       "8        OFALOU+ErasUltra  11.4    .\n",
       "9   OFALOU+Helvetica-Bold   9.4     \n",
       "10  OFALOU+Frutiger-Black   9.4    –\n",
       "11  OFALOU+Frutiger-Black   9.4     \n",
       "12  OFALOU+Frutiger-Black   9.4    O\n",
       "13  OFALOU+Frutiger-Black   9.4    B\n",
       "14  OFALOU+Frutiger-Black   9.4    J\n",
       "15  OFALOU+Frutiger-Black   9.4    E\n",
       "16  OFALOU+Frutiger-Black   9.4    C\n",
       "17  OFALOU+Frutiger-Black   9.4    T\n",
       "18  OFALOU+Frutiger-Black   9.4    I\n",
       "19  OFALOU+Frutiger-Black   9.4    V\n",
       "20  OFALOU+Frutiger-Black   9.4    E\n",
       "21  OFALOU+Frutiger-Black   9.4    :\n",
       "22  OFALOU+Helvetica-Bold   9.4     \n",
       "23  OFALOU+Helvetica-Bold   9.4    M\n",
       "24  OFALOU+Helvetica-Bold   9.4    i\n",
       "25  OFALOU+Helvetica-Bold   9.4    t\n",
       "26  OFALOU+Helvetica-Bold   9.4    o\n",
       "27  OFALOU+Helvetica-Bold   9.4    c\n",
       "28  OFALOU+Helvetica-Bold   9.4    h\n",
       "29  OFALOU+Helvetica-Bold   9.4    o\n",
       "30  OFALOU+Helvetica-Bold   9.4    n\n",
       "31  OFALOU+Helvetica-Bold   9.4    d\n",
       "32  OFALOU+Helvetica-Bold   9.4    r\n",
       "33  OFALOU+Helvetica-Bold   9.4    i\n",
       "34  OFALOU+Helvetica-Bold   9.4    a\n",
       "35  OFALOU+Helvetica-Bold   9.4     \n",
       "36  OFALOU+Helvetica-Bold   9.4    a\n",
       "37  OFALOU+Helvetica-Bold   9.4    r\n",
       "38  OFALOU+Helvetica-Bold   9.4    e\n",
       "39  OFALOU+Helvetica-Bold   9.4     \n",
       "40  OFALOU+Helvetica-Bold   9.4    a\n",
       "41  OFALOU+Helvetica-Bold   9.4    b\n",
       "42  OFALOU+Helvetica-Bold   9.4    u\n",
       "43  OFALOU+Helvetica-Bold   9.4    n\n",
       "44  OFALOU+Helvetica-Bold   9.4    d\n",
       "45  OFALOU+Helvetica-Bold   9.4    a\n",
       "46  OFALOU+Helvetica-Bold   9.4    n\n",
       "47  OFALOU+Helvetica-Bold   9.4    t\n",
       "48  OFALOU+Helvetica-Bold   9.4     \n",
       "49  OFALOU+Helvetica-Bold   9.4    i\n",
       "50  OFALOU+Helvetica-Bold   9.4    n\n",
       "51  OFALOU+Helvetica-Bold   9.4     \n",
       "52  OFALOU+Helvetica-Bold   9.4    l\n",
       "53  OFALOU+Helvetica-Bold   9.4    i\n",
       "54  OFALOU+Helvetica-Bold   9.4    v\n",
       "55  OFALOU+Helvetica-Bold   9.4    e\n",
       "56  OFALOU+Helvetica-Bold   9.4    r\n",
       "57  OFALOU+Helvetica-Bold   9.4    .\n",
       "58  OFALOU+Helvetica-Bold   9.4     \n",
       "59  OFALOU+Helvetica-Bold   9.4    T\n",
       "60  OFALOU+Helvetica-Bold   9.4    h\n",
       "61  OFALOU+Helvetica-Bold   9.4    e\n",
       "62  OFALOU+Helvetica-Bold   9.4     \n",
       "63  OFALOU+Helvetica-Bold   9.4    r\n",
       "64  OFALOU+Helvetica-Bold   9.4    o\n",
       "65  OFALOU+Helvetica-Bold   9.4    l\n",
       "66  OFALOU+Helvetica-Bold   9.4    e\n",
       "67  OFALOU+Helvetica-Bold   9.4    s\n",
       "68  OFALOU+Helvetica-Bold   9.4     \n",
       "69  OFALOU+Helvetica-Bold   9.4    o\n",
       "70  OFALOU+Helvetica-Bold   9.4    f\n",
       "71  OFALOU+Helvetica-Bold   9.4     \n",
       "72  OFALOU+Helvetica-Bold   9.4    m\n",
       "73  OFALOU+Helvetica-Bold   9.4    i\n",
       "74  OFALOU+Helvetica-Bold   9.4    t\n",
       "75  OFALOU+Helvetica-Bold   9.4    o\n",
       "76  OFALOU+Helvetica-Bold   9.4    c\n",
       "77  OFALOU+Helvetica-Bold   9.4    h\n",
       "78  OFALOU+Helvetica-Bold   9.4    o\n",
       "79  OFALOU+Helvetica-Bold   9.4    n\n",
       "80  OFALOU+Helvetica-Bold   9.4    d\n",
       "81  OFALOU+Helvetica-Bold   9.4    r\n",
       "82  OFALOU+Helvetica-Bold   9.4    i\n",
       "83  OFALOU+Helvetica-Bold   9.4    a\n",
       "84  OFALOU+Helvetica-Bold   9.4    l\n",
       "85  OFALOU+Helvetica-Bold   9.4     \n",
       "86  OFALOU+Helvetica-Bold   9.4    p\n",
       "87  OFALOU+Helvetica-Bold   9.4    r\n",
       "88  OFALOU+Helvetica-Bold   9.4    o\n",
       "89  OFALOU+Helvetica-Bold   9.4    t\n",
       "90  OFALOU+Helvetica-Bold   9.4    e\n",
       "91  OFALOU+Helvetica-Bold   9.4    i\n",
       "92  OFALOU+Helvetica-Bold   9.4    n\n",
       "93  OFALOU+Helvetica-Bold   9.4     \n",
       "94  OFALOU+Helvetica-Bold   9.4    i\n",
       "95  OFALOU+Helvetica-Bold   9.4    n\n",
       "96  OFALOU+Helvetica-Bold   9.4     \n",
       "97  OFALOU+Helvetica-Bold   9.4    l\n",
       "98  OFALOU+Helvetica-Bold   9.4    i\n",
       "99  OFALOU+Helvetica-Bold   9.4    v"
      ]
     },
     "execution_count": 103,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import numpy as np\n",
    "import pandas as pd\n",
    "all_size = [round(x['size'], 1) for x in left_crop.chars]\n",
    "all_font = [x['fontname'] for x in left_crop.chars]\n",
    "all_text = [x['text'] for x in left_crop.chars]\n",
    "df_size = pd.DataFrame({\n",
    "    'fontname': all_font,\n",
    "    'size': all_size,\n",
    "    'text': all_text\n",
    "    })\n",
    "df_size[:100]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "article = Article('Mitochondrial protein UCP1 mediates liver injury induced by LPS through EKR signaling pathway')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "article.parse(left_crop)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Article Name Mitochondrial protein UCP1 mediates liver injury induced by LPS through EKR signaling pathway\n",
      "# All Chapters ['Abstract.']\n",
      "1## Chapter Name Abstract.\n",
      "   All Sections ['Head Section', '– OBJECTIVE:', 'MATERIALS AND METHODS:', 'RESULTS:', 'CONCLUSIONS: ']\n",
      "1### Section Name Head Section\n",
      "     \n",
      "2### Section Name – OBJECTIVE:\n",
      "     Mitochondria are abundant in liver. The roles of mitochondrial protein in liver injury and related signaling path-ways are still unclear. UCP1 is a novel mitochon-drial transmembrane protein. Its expression pattern and function in liver still needs further investigation.\n",
      "3### Section Name MATERIALS AND METHODS:\n",
      "     A mouse mod-el of liver injury was established by the treat-ment of LPS. UCP1 expression in the liver tis-sue was detected by Western blot and qRT-PCR. ERK signaling activity was tested by enzymat-ic activity kit. ATP production was evaluated by flow cytometry. Cell apoptosis was determined by Western blot and flow cytometry. ERK signal-ing pathway inhibitor, U0126, was used to pre-treat mice. Liver tissue from sepsis patients was collected from the surgery.\n",
      "4### Section Name RESULTS:\n",
      "     Our data showed that the level of UCP1 was upregulated, ERK signaling was ac-tivated, ATP production was reduced, and cell apoptosis was enhanced in mice with liver in-jury model caused by LPS. U0126 intervention significantly suppressed UCP1 expression, in-hibited ERK signaling pathway, enhanced ATP production, and restrained liver cell apoptosis in mice liver injury model. UCP1 increased, ERK signaling activated, and cell apoptosis elevated in the liver tissue of sepsis patients.\n",
      "5### Section Name CONCLUSIONS: \n",
      "    UCP1 plays a role in the liv-er tissue of mouse liver injury model and sepsis patients through the modulation of mitochon-drial ATP production and cell apoptosis by ERK signaling pathway.\n"
     ]
    }
   ],
   "source": [
    "article.print_chapters()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "article.parse(right_crop)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Article Name Mitochondrial protein UCP1 mediates liver injury induced by LPS through EKR signaling pathway\n",
      "# All Chapters ['Abstract.', 'Introduction', 'Materials and Methods', 'Results']\n",
      "1## Chapter Name Abstract.\n",
      "   All Sections ['Head Section', '– OBJECTIVE:', 'MATERIALS AND METHODS:', 'RESULTS:', 'CONCLUSIONS: ']\n",
      "1### Section Name Head Section\n",
      "     \n",
      "2### Section Name – OBJECTIVE:\n",
      "     Mitochondria are abundant in liver. The roles of mitochondrial protein in liver injury and related signaling path-ways are still unclear. UCP1 is a novel mitochon-drial transmembrane protein. Its expression pattern and function in liver still needs further investigation.\n",
      "3### Section Name MATERIALS AND METHODS:\n",
      "     A mouse mod-el of liver injury was established by the treat-ment of LPS. UCP1 expression in the liver tis-sue was detected by Western blot and qRT-PCR. ERK signaling activity was tested by enzymat-ic activity kit. ATP production was evaluated by flow cytometry. Cell apoptosis was determined by Western blot and flow cytometry. ERK signal-ing pathway inhibitor, U0126, was used to pre-treat mice. Liver tissue from sepsis patients was collected from the surgery.\n",
      "4### Section Name RESULTS:\n",
      "     Our data showed that the level of UCP1 was upregulated, ERK signaling was ac-tivated, ATP production was reduced, and cell apoptosis was enhanced in mice with liver in-jury model caused by LPS. U0126 intervention significantly suppressed UCP1 expression, in-hibited ERK signaling pathway, enhanced ATP production, and restrained liver cell apoptosis in mice liver injury model. UCP1 increased, ERK signaling activated, and cell apoptosis elevated in the liver tissue of sepsis patients.\n",
      "5### Section Name CONCLUSIONS: \n",
      "    UCP1 plays a role in the liv-er tissue of mouse liver injury model and sepsis patients through the modulation of mitochon-drial ATP production and cell apoptosis by ERK signaling pathway.\n",
      "2## Chapter Name Introduction\n",
      "   All Sections ['Head Section']\n",
      "1### Section Name Head Section\n",
      "    LPS induced liver injury, which belongs to a kind of liver inflammatory response syndrome. It is cha-racterized by high incidence, high case fatality rate, high cost, and increasing incidence. Liver injury caused by LPS can lead to tissue hypoperfusion, septic shock, organ dysfunction, and death throu-gh multiple mechanisms. Therefore, effective de-tection is the premise and foundation of the treat-ment of liver injury induced by LPS. According to the current criteria, the treatment to the mice with severe inflammation became less effective. At pre-sent, it is urgently needed to explore the biomarker with good specificity and sensitivity for early dia-gnosis, treatment, and prognosis of LPS-induced liver injury.Previous studies showed that biomar-kers for LPS-induced liver injury mainly included the indicator of tissue perfusion, organ function, inflammatory cytokines, and liver tissue dynamics. However, the limitation was that all of these indica-tors were lack of specificity and high sensitivity. Therefore, a better molecular biomarker is needed for the diagnosis of liver injury caused by LPS in clinic.LPS-induced liver injury involves a variety of mechanisms, including the structure and function changes of mitochondria associated with inflam-matory mediators. Myocardial mitochondria are the important target of liver cell during liver injury. Following the development of biogenetics and the concept of cellular hypoxia, it was speculated that the mitochondrial energy metabolism had certain relationship with its own dynamics changes. Mi-tochondrial protein UCP played an important role in mitochondrial function, which may be closely related to liver injury. In this study, we intend to explore the potential relationship between UCP1 le-vels in liver cells and liver injury induced by LPS.As there is still lack of evidence about the relation-ship between UCP1 and LPS induced liver injury, we selected patients with liver injury from ICU and tested UCP1 level in the liver based on the data con-cerning mouse model of liver injury, in order to eva-luate the value of UCP1 as molecular biomarker in liver injury induced by LPS and provide theoretical basis for the early diagnosis and prognosis of liver injury.\n",
      "3## Chapter Name Materials and Methods\n",
      "   All Sections ['Experimental Model', 'Liver Tissue Preparation', 'Caspase-3 Activity Detection', 'ATP Production', 'Western Blot', 'Real-time PCR', 'U0126 Pretreatment', 'Statistical Analysis']\n",
      "1### Section Name Experimental Model\n",
      "    Mouse liver injury model was established using LPS according to previous report. Female C57BL/6 mouse at 6-8 weeks old was intraperi-toneal injected by LPS at 25 μg/kg. The mouse injected by normal saline was treated as control. The experiment was approved by the Ethics Com-mittee in our hospital conforming to animal wel-fare. The liver tissue collected from the sepsis patients received surgery was used as experimen-tal group, while the liver tissue obtained from li-ver operation due to traffic accident was treated as control. The experiment was approved by the Ethics Committee in Qingdao Municipal Hospi-tal. All the subjects had signed informed consent.\n",
      "2### Section Name Liver Tissue Preparation\n",
      "    The liver tissue was extracted from the mouse or operation. The tissue was lysed and washed by normal saline and acetone for four times. Then, the homogenate was centrifuged at 150 g and sto-red at -20°C.\n",
      "3### Section Name Caspase-3 Activity Detection\n",
      "    ERK signaling activity was detected by kinase activity kit. The caspase-3 activity was tested by microplate reader. A total of 2 × 10 liver cel-ls were added with lysate to release the protein. Then, the mixture was incubated with caspase-3 chromophoric substance at room temperature avoid of light for 20 min. At last, cells were tested on microplate reader at 492 nm.\n",
      "4### Section Name ATP Production\n",
      "    Mitochondrial ATP production in liver tissue was detected by flow cytometry. The liver tissue was added with lysis to prepare the suspension. Next, the mixture was added with detection flu-id and tested on microplate reader at 560 nm to calculate the ATP concentration according to the standard curve.\n",
      "5### Section Name Western Blot\n",
      "    The protein was extracted from the liver tissue and quantified. Next, the protein was separated by SDS-PAGE and analyzed by ImageJ software to obtain the gray value.\n",
      "6### Section Name Real-time PCR\n",
      "    Total RNA was extracted from the liver tissue using Trizol (Invitrogen, Carlsbad, CA, USA) and reverse transcribed for Real-time PCR. To-tal RNA was reverse transcribed to cDNA using polyA at 42°C for 60 min. The PCR reaction system contained 1 μl cDNA, 2 μl primers, 4 μl dNTP, 2 μl loading buffer, and 1 μl Taq enzyme. The PCR reaction was composed of 94°C for 5 min, followed by 30 cycles of 94°C for 30 s and 55°C for 30 s. The primer sequences were as follows.UCP1, 5’-TTATTAGAGGGTGGGGTGGATTGT-3’ and 5’- CCACCTAAATCAACCTCCAACCA-3’; actin, 5’-TTATTAGAGGGTGGGGCGGATCGC-3’ and 5’- CCACCTAAATCGACCTCCGACCG-3’.\n",
      "7### Section Name U0126 Pretreatment\n",
      "    ERK signaling pathway inhibitor U0126 was used to treat mouse through intraperitoneal injection at 10 μg/kg. Next, the mouse was intra-peritoneal injected by LPS at 25 μg/kg. The mouse injected by normal saline was treated as control.\n",
      "8### Section Name Statistical Analysis\n",
      "    All data analyses were performed on SPSS 11.0 (SPSS Inc., Chicago, IL, USA). The measurement data were depicted as mean ± standard deviation. p<0.05 was considered as statistical significance.\n",
      "4## Chapter Name Results\n",
      "   All Sections ['UCP1 Expression in Liver Cells from Mouse Liver Injury Model']\n",
      "1### Section Name UCP1 Expression in Liver Cells from Mouse Liver Injury Model\n",
      "    As shown in Figure 1A, qRT-PCR results demonstrated that UCP1 mRNA significant-ly increased in liver cells from liver injury model compared with that in normal control (p=0.0017). As shown in Figure 1B and C, We-stern blot results revealed that the level of UCP1 protein obviously elevated in liver cells from li-ver injury model compared with that in normal control (p=0.0027).\n"
     ]
    }
   ],
   "source": [
    "article.print_chapters()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "article = Article('Mitochondrial protein UCP1 mediates liver injury induced by LPS through EKR signaling pathway')\n",
    "for page in pdf.pages:\n",
    "    x_edge = 0.07\n",
    "    y_edge = 0.1\n",
    "    half = 0.5\n",
    "    height = page.height\n",
    "    width = page.width\n",
    "    left_crop = page.within_bbox((width * x_edge, height * y_edge, width * half, height * (1 - y_edge)))\n",
    "    right_crop = page.within_bbox((width * half, height * y_edge, width * (1 - x_edge), height * (1 - y_edge)))\n",
    "    article.parse(left_crop)\n",
    "    article.parse(right_crop)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "# Mitochondrial protein UCP1 mediates liver injury induced by LPS through EKR signaling pathway\n",
      "## Abstract.\n",
      "### Head Section\n",
      " \n",
      "### – OBJECTIVE:\n",
      " Mitochondria are abundant in liver. The roles of mitochondrial protein in liver injury and related signaling path-ways are still unclear. UCP1 is a novel mitochon-drial transmembrane protein. Its expression pattern and function in liver still needs further investigation.\n",
      "### MATERIALS AND METHODS:\n",
      " A mouse mod-el of liver injury was established by the treat-ment of LPS. UCP1 expression in the liver tis-sue was detected by Western blot and qRT-PCR. ERK signaling activity was tested by enzymat-ic activity kit. ATP production was evaluated by flow cytometry. Cell apoptosis was determined by Western blot and flow cytometry. ERK signal-ing pathway inhibitor, U0126, was used to pre-treat mice. Liver tissue from sepsis patients was collected from the surgery.\n",
      "### RESULTS:\n",
      " Our data showed that the level of UCP1 was upregulated, ERK signaling was ac-tivated, ATP production was reduced, and cell apoptosis was enhanced in mice with liver in-jury model caused by LPS. U0126 intervention significantly suppressed UCP1 expression, in-hibited ERK signaling pathway, enhanced ATP production, and restrained liver cell apoptosis in mice liver injury model. UCP1 increased, ERK signaling activated, and cell apoptosis elevated in the liver tissue of sepsis patients.\n",
      "### CONCLUSIONS: \n",
      "UCP1 plays a role in the liv-er tissue of mouse liver injury model and sepsis patients through the modulation of mitochon-drial ATP production and cell apoptosis by ERK signaling pathway.\n",
      "## Introduction\n",
      "### Head Section\n",
      "LPS induced liver injury, which belongs to a kind of liver inflammatory response syndrome. It is cha-racterized by high incidence, high case fatality rate, high cost, and increasing incidence. Liver injury caused by LPS can lead to tissue hypoperfusion, septic shock, organ dysfunction, and death throu-gh multiple mechanisms. Therefore, effective de-tection is the premise and foundation of the treat-ment of liver injury induced by LPS. According to the current criteria, the treatment to the mice with severe inflammation became less effective. At pre-sent, it is urgently needed to explore the biomarker with good specificity and sensitivity for early dia-gnosis, treatment, and prognosis of LPS-induced liver injury.Previous studies showed that biomar-kers for LPS-induced liver injury mainly included the indicator of tissue perfusion, organ function, inflammatory cytokines, and liver tissue dynamics. However, the limitation was that all of these indica-tors were lack of specificity and high sensitivity. Therefore, a better molecular biomarker is needed for the diagnosis of liver injury caused by LPS in clinic.LPS-induced liver injury involves a variety of mechanisms, including the structure and function changes of mitochondria associated with inflam-matory mediators. Myocardial mitochondria are the important target of liver cell during liver injury. Following the development of biogenetics and the concept of cellular hypoxia, it was speculated that the mitochondrial energy metabolism had certain relationship with its own dynamics changes. Mi-tochondrial protein UCP played an important role in mitochondrial function, which may be closely related to liver injury. In this study, we intend to explore the potential relationship between UCP1 le-vels in liver cells and liver injury induced by LPS.As there is still lack of evidence about the relation-ship between UCP1 and LPS induced liver injury, we selected patients with liver injury from ICU and tested UCP1 level in the liver based on the data con-cerning mouse model of liver injury, in order to eva-luate the value of UCP1 as molecular biomarker in liver injury induced by LPS and provide theoretical basis for the early diagnosis and prognosis of liver injury.\n",
      "## Materials and Methods\n",
      "### Experimental Model\n",
      "Mouse liver injury model was established using LPS according to previous report. Female C57BL/6 mouse at 6-8 weeks old was intraperi-toneal injected by LPS at 25 μg/kg. The mouse injected by normal saline was treated as control. The experiment was approved by the Ethics Com-mittee in our hospital conforming to animal wel-fare. The liver tissue collected from the sepsis patients received surgery was used as experimen-tal group, while the liver tissue obtained from li-ver operation due to traffic accident was treated as control. The experiment was approved by the Ethics Committee in Qingdao Municipal Hospi-tal. All the subjects had signed informed consent.\n",
      "### Liver Tissue Preparation\n",
      "The liver tissue was extracted from the mouse or operation. The tissue was lysed and washed by normal saline and acetone for four times. Then, the homogenate was centrifuged at 150 g and sto-red at -20°C.\n",
      "### Caspase-3 Activity Detection\n",
      "ERK signaling activity was detected by kinase activity kit. The caspase-3 activity was tested by microplate reader. A total of 2 × 10 liver cel-ls were added with lysate to release the protein. Then, the mixture was incubated with caspase-3 chromophoric substance at room temperature avoid of light for 20 min. At last, cells were tested on microplate reader at 492 nm.\n",
      "### ATP Production\n",
      "Mitochondrial ATP production in liver tissue was detected by flow cytometry. The liver tissue was added with lysis to prepare the suspension. Next, the mixture was added with detection flu-id and tested on microplate reader at 560 nm to calculate the ATP concentration according to the standard curve.\n",
      "### Western Blot\n",
      "The protein was extracted from the liver tissue and quantified. Next, the protein was separated by SDS-PAGE and analyzed by ImageJ software to obtain the gray value.\n",
      "### Real-time PCR\n",
      "Total RNA was extracted from the liver tissue using Trizol (Invitrogen, Carlsbad, CA, USA) and reverse transcribed for Real-time PCR. To-tal RNA was reverse transcribed to cDNA using polyA at 42°C for 60 min. The PCR reaction system contained 1 μl cDNA, 2 μl primers, 4 μl dNTP, 2 μl loading buffer, and 1 μl Taq enzyme. The PCR reaction was composed of 94°C for 5 min, followed by 30 cycles of 94°C for 30 s and 55°C for 30 s. The primer sequences were as follows.UCP1, 5’-TTATTAGAGGGTGGGGTGGATTGT-3’ and 5’- CCACCTAAATCAACCTCCAACCA-3’; actin, 5’-TTATTAGAGGGTGGGGCGGATCGC-3’ and 5’- CCACCTAAATCGACCTCCGACCG-3’.\n",
      "### U0126 Pretreatment\n",
      "ERK signaling pathway inhibitor U0126 was used to treat mouse through intraperitoneal injection at 10 μg/kg. Next, the mouse was intra-peritoneal injected by LPS at 25 μg/kg. The mouse injected by normal saline was treated as control.\n",
      "### Statistical Analysis\n",
      "All data analyses were performed on SPSS 11.0 (SPSS Inc., Chicago, IL, USA). The measurement data were depicted as mean ± standard deviation. p<0.05 was considered as statistical significance.\n",
      "## Results\n",
      "### UCP1 Expression in Liver Cells from Mouse Liver Injury Model\n",
      "As shown in Figure 1A, qRT-PCR results demonstrated that UCP1 mRNA significant-ly increased in liver cells from liver injury model compared with that in normal control (p=0.0017). As shown in Figure 1B and C, We-stern blot results revealed that the level of UCP1 protein obviously elevated in liver cells from li-ver injury model compared with that in normal control (p=0.0027).\n",
      "### ERK Signaling Activated, Mitochondrial ATP Production Reduced, and Cell Apoptosis Enhanced in Liver Injury Mouse\n",
      "To further discuss the molecular mechanism of UCP1 in LPS induced liver injury model, we tested mitochondrial ATP production and cell apoptosis. As shown in Figure 2, our data exhibited that ERK signaling pathway was activated, mitochondrial ATP production was reduced, and cell apoptosis was enhanced in mice with liver injury. \n",
      "### The Influence of ERK Signaling Pathway Inhibitor U0126 Pretreatment\n",
      "Next, to investigate the specific mechanism of UCP1 in liver injury induced by LPS, we detected ATP level and cell apoptosis in mice model pretre-ated by U0126. As shown in Figure 3, U0126 inter-vention significantly suppressed UCP1 expression, inhibited ERK signaling pathway, enhanced ATP production, and restrained liver cell apoptosis in mice, suggesting a key role of U0126 on ERK si-gnaling pathway.\n",
      "### UCP1 Expression and Cell Apoptosis in the Liver Tissue from Sepsis Patients\n",
      "To study UCP1 expression and cell apoptosis in sepsis patients, we collected the liver tissue from sepsis patients who received surgery. As shown in Figure 4, the level of UCP1 was increased, along with the increasing activation of ERK signaling pathway and elevation of cell apoptosis in the li-ver tissue of sepsis patients.\n",
      "## Discussion\n",
      "### Head Section\n",
      "LPS seriously threatens mammal’s life via the cause of liver inflammation, among which, the function of mitochondrial uncoupling protein (UCP1) in liver injury induced by LPS remains to be further investigated. Early diagnosis and treat-ment were of prime importance for liver injury in-duced by LPS, but current molecular biomar-ker provided an inconvenient and unsatisfactory solution to evaluate this disease. Therefore, this study aimed to explore the possibility of UCP1 as the biomarker in early diagnosis and prognosis of liver injury induced by LPS. LPS induced liver injury is a complicated inflammatory response process induced by infection. A variety of in-flammatory cytokines interact with each other to induce intense immune response during the pro-cess of liver injury, thus leading to mouse dea-th. Under this circumstance, pro-inflammatory cytokines signaling pathways were activated to lead to the damage in organs and tissues of mam-mal and anti- anti-inflammatory strategy should be adopted to prevent inflammatory diseases. Therefore, the application of reasonable detection method for early detection and prognosis is of great significance. In our study on the molecu-lar mechanism of UCP1 in liver injury, we found that UCP1 was upregulated, ERK signaling was activated, ATP production was reduced, and cell apoptosis enhanced in mouse model with liver injury induced by LPS. Also, with the treatment of U0126, the expression of UCP1 was significantly suppressed, ERK signaling pathway was inhibi-ted, ATP production was increased, and liver cell apoptosis was decreased in mice. It suggested that UCP1 played its role in liver injury through the regulation of ERK signaling pathway, ATP pro-duction and cell apoptosis. Endotoxin injection in mouse showed the production of reactive oxygen species (ROS), sustained inflammatory activa-tion and over expression of immune cytokines. Moreover, levels of inflammatory cytokines were upregulated when ROS was activated. Thus, the role of inflammatory cytokines in regulating reactive oxygen species and the influence of mole-cular target still needs further exploration. There are also deficiencies in this study. Firstly, the sam-ple scale was small and larger scale was needed to investigate the possibility of UCP1 as a practical biomarker in liver injury induced by LPS. Se-condly, since most liver injury patients received chemotherapy and other treatments, the impact of treatment on UCP1 level is still to be further elucidated. Lastly, UCP1 as the target in animal liver injury model requires to be evaluated. \n",
      "## Conclusions\n",
      "### Head Section\n",
      "UCP1 exerted an essential effect on mice with liver injury and sepsis patients by regulating mi-tochondrial ATP production, cell apoptosis and ERK signaling pathway.\n",
      "## References\n"
     ]
    }
   ],
   "source": [
    "article.print_chapters()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3.8.12 ('cv')",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.12 (default, Oct 12 2021, 13:49:34) \n[GCC 7.5.0]"
  },
  "orig_nbformat": 4,
  "vscode": {
   "interpreter": {
    "hash": "ea9525bd9fde1ba6002997797d728afbfc4fa59e53537b36ac1765c9b353d88b"
   }
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
